ambry-frontend/src/main/java/com/github/ambry/frontend/AmbryIdConverterFactory.java

```diff
@@ -108,7 +109,8 @@ public Future<String> convert(RestRequest restRequest, String input, BlobInfo bl
       try {
         if (!isOpen) {
           exception = new RestServiceException("IdConverter is closed", RestServiceErrorCode.ServiceUnavailable);
-        } else if (restRequest.getRestMethod().equals(RestMethod.POST)) {
+        } else if (restRequest.getRestMethod().equals(RestMethod.POST) && !RestUtils.isS3Request(restRequest)) {
+          // Ambry chunk uploads come as POST requests. Sign id if required.
```

Comment:
Sure. I am just trying to figure out value of `ambry-internal-key-signed-id-metadata` for S3 multipart uploads. I added it as a TODO item for next PR.

Reply:
if we sign the s3 blob id, which is already a serialized json object, we would add signedId/ prefix to the final signed string, which means in the Etag, we returned back to client, it would look something like /signedId/xxxxxxxxxxx. I think we might need to sign s3 etag in a different way.

Reply:
Sure. Yeah, true. We might need to sign S3 ETag in a different way. I will work on it in the next PR.